frontend/src/components/CredentialTile.vue

```diff
@@ -0,0 +1,157 @@
+    <v-list-item-action :class="{'py-0 my-1' : dense}">
+      <rotate-credentials :shoot-item="shootItem" :type="type"></rotate-credentials>
+    </v-list-item-action>
+  </v-list-item>
+</template>
+
+<script>
+import RotateCredentials from '@/components/RotateCredentials'
+import TimeString from '@/components/TimeString'
+import ShootMessages from '@/components/ShootMessages/ShootMessages'
+import { shootItem } from '@/mixins/shootItem'
+import get from 'lodash/get'
+import flatMap from 'lodash/flatMap'
+import head from 'lodash/head'
+import compact from 'lodash/compact'
+import filter from 'lodash/filter'
+import find from 'lodash/find'
+import { rotationTypes } from '@/utils/credentialsRotation'
+
+export default {
+  name: 'credential-tile',
+  components: {
+    RotateCredentials,
+    TimeString,
+    ShootMessages
+  },
+  props: {
+    type: {
+      type: String,
+      required: false
+    },
+    dense: {
+      type: Boolean,
+      required: false
+    }
+  },
+  mixins: [shootItem],
+  computed: {
+    rotationStatus () {
+      return get(this.shootStatusCredentialsRotation, this.type, {})
+    },
+    lastInitiationTime () {
+      if (this.type) {
+        return this.rotationStatus.lastInitiationTime
+      }
+      // Do not show aggregated initiation time
+      return undefined
+    },
+    lastCompletionTime () {
+      if (this.type) {
+        return this.rotationStatus.lastCompletionTime
+      }
+      const allCompletionTimestamps = compact(flatMap(this.shootStatusCredentialsRotation, 'lastCompletionTime')).sort()
+      let requiredNumberOfRotationTimestamps = filter(rotationTypes, 'type').length // only consider rotations that have an rotation status type
+      if (!this.shootEnableStaticTokenKubeconfig) {
+        requiredNumberOfRotationTimestamps = requiredNumberOfRotationTimestamps - 1
+      }
+
+      if (requiredNumberOfRotationTimestamps === allCompletionTimestamps.length) {
+        return head(allCompletionTimestamps)
+      }
+      return undefined
+    },
+    phase () {
+      if (!this.type) {
+        return this.shootStatusCredentialsRotationAggregatedPhase
+      }
+      const type = this.rotationStatus.phase
+      return {
+        type
+      }
+    },
+    phaseCaption () {
+      return get(this.phase, 'caption', this.phase.type)
+    },
+    phaseColor () {
+      switch (this.phase.type) {
+        case 'Prepared':
+        case 'Completed':
+          return 'primary'
+        default:
+          return 'info'
+      }
+    },
+    iconColor () {
+      if (!this.isCACertificateValiditiesAcceptable) {
+        return 'warning'
+      }
+      return 'primary'
+    },
+    showChip () {
+      return this.phase.type && this.phase.type !== 'Completed'
+    },
+    title () {
+      return this.getRotationTitle(this.type)
+    }
+  },
+  methods: {
+    getRotationTitle (type) {
+      return get(find(rotationTypes, t => { return t.type === this.type }), 'title')
+    }
```

Comment:
do we really need to extract this into a method? can't we inline it in the getter?
in addition, the passed `type` parameter is not used

Reply:
And you do not even use the parameter `type` of the method.
```js
    title () {
      const rotationType = find(rotationTypes, ['type', this.type])
      return rotationType?.title
    }
```

Reply:
The `type` param is not required, I removed it.
I moved the method body into the getter were we use it (as Holger suggested).
The reason why I use a method for the `find` is because `type` can be `undefined`. In case of `all credentials` there is no type, see also comment in `credentialsRotation.js`:
```
// no type as this rotation is the 'all credential rotatiom trigger', which as no dedicated key in the rotation status
```
find with the shorthand syntax does not match `undefined`

Reply:
I had an 'artificial' type for allCredentials but removed it due to the feedback. However, I'm not 100% sure if this improved the code.

Reply:
re-introduced type for `all credentials`